src/dev_wasi_nn_nif.erl

```diff
@@ -0,0 +1,373 @@
+%%% @doc WASI-NN NIF module for HyperBEAM.
+%%% Implements native functions for AI model loading and inference.
+%%% This module provides the NIF interface for the dev_wasi_nn module.
+-module(dev_wasi_nn_nif).
+-include("include/hb.hrl").
+-include_lib("eunit/include/eunit.hrl").
+-export([
+    init_backend/0,
+    load_by_name_with_config/3,
+    init_execution_context/2,
+    close_execution_context/2,
+    deinit_backend/1,
+    run_inference/3
+]).
+-export([init_execution_context_once/2, switch_model/2]).
+-export([cleanup_model_contexts/1, cleanup_all_contexts/0, get_current_model_info/0]).
+
+-on_load(init/0).
+-define(NOT_LOADED, not_loaded(?LINE)).
+
+%% Module-level cache
+-define(CACHE_TAB, wasi_nn_cache).
+-define(SINGLETON_KEY, global_cache).
+-define(CACHE_OWNER_NAME, wasi_nn_cache_owner).  % Registered name for cache owner process
+
+%% @doc Start the dedicated ETS table owner process.
+%% Creates a persistent process that owns the ETS table to ensure it remains
+%% available even if the calling process terminates.
+%%
+%% @returns {ok, Pid} where Pid is the process ID of the cache owner.
+start_cache_owner() ->
+    case whereis(?CACHE_OWNER_NAME) of
+        undefined ->
+            % No owner process exists, create one
+            Pid = spawn(fun() -> 
+                % Create the table if it doesn't exist
+                case ets:info(?CACHE_TAB) of
+                    undefined ->
+                        ?event(dev_wasi_nn_nif, {cache_owner_creating_table, ?CACHE_TAB}),
+                        ets:new(?CACHE_TAB, [set, named_table, public]);
+                    _ ->
+                        ?event(dev_wasi_nn_nif, {cache_table_already_exists, ?CACHE_TAB})
+                end,
+                % Register the process with a name for easy lookup
+                register(?CACHE_OWNER_NAME, self()),
+                cache_owner_loop()
+            end),
+            {ok, Pid};
+        Pid ->
+            % Owner process already exists
+            {ok, Pid}
+    end.
+
+%% @doc Loop function for the cache owner process.
+%% Keeps the process alive to maintain ownership of the ETS table.
+%% Handles stop messages and ping requests.
+%%
+%% @returns ok when the process is stopped.
+cache_owner_loop() ->
+    receive
+        stop -> 
+            ?event(dev_wasi_nn_nif, {cache_owner_stopping}),
+            ok;
+        {From, ping} ->
+            From ! {self(), pong},
+            cache_owner_loop();
+        _ -> 
+            cache_owner_loop()
+    after 
+        3600000 -> % Stay alive for a long time (1 hour), then check again
```

Comment:
Out of interest, what is the motivation for this? It doesn't hurt, but I can't see intuitively how it helps either?